ios-app/UI/ZoomMeetViewController.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  ZoomMeetViewController.swift
+//  ios-app
+//
+//  Copyright © 2020 Testpress. All rights reserved.
+//
+
+import UIKit
+import MobileRTC
+import MobileCoreServices
+
+
+class ZoomMeetViewController: UIViewController, MobileRTCAuthDelegate, MobileRTCMeetingServiceDelegate, UIAdaptivePresentationControllerDelegate {
+    
+    var activityIndicator: UIActivityIndicatorView!
+    var accessToken: String!
+    var meetingNumber: String!
+    var password: String!
+    var meetingTitle: String!
+    var hasError: Bool = false
+    var emptyView: EmptyView!
+    var fetchAccessToken: ((_ completion: @escaping(String?, TPError?) -> Void) -> Void)!
+    @IBOutlet weak var containerView: UIStackView!
+    
+    override func viewDidLoad() {
+        self.setStatusBarColor()
+        initializeLoadingScreen()
+        emptyView = EmptyView.getInstance(parentView: containerView)
+        self.initialize()
+    }
+    
+    func initializeLoadingScreen() {
+        activityIndicator = UIUtils.initActivityIndicator(parentView: self.view)
+        activityIndicator?.center = CGPoint(x: view.center.x, y: view.center.y + 70)
+        let pagingSpinner = UIActivityIndicatorView(style: .gray)
+        pagingSpinner.startAnimating()
+        pagingSpinner.color = Colors.getRGB(Colors.PRIMARY)
+        pagingSpinner.hidesWhenStopped = true
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        activityIndicator?.startAnimating()
+        openPreviousPageIfBackButtonPressedFromZoom()
+    }
+    
+    func openPreviousPageIfBackButtonPressedFromZoom() {
+        if let klass = NSClassFromString("ZMNavigationController") {
+            if self.presentedViewController?.isKind(of: klass) ?? false{
+                self.gotoPreviousPage()
+            }
+        }
+    }
+    
+    func initialize() {
+        let zoomSDK = MobileRTCSDKInitContext()
+        zoomSDK.domain = "zoom.us"
+        zoomSDK.enableLog = true
+        MobileRTC.shared().initialize(zoomSDK)
+        let authService = MobileRTC.shared().getAuthService()
+        authService?.delegate = self
+        authService?.jwtToken = accessToken!
+        authService?.sdkAuth()
+    }
+    
+    @IBAction func back(_ sender: Any) {
+        gotoPreviousPage()
+    }
+    
+    func gotoPreviousPage() {
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        let viewController = storyboard.instantiateViewController(withIdentifier:
+            Constants.TAB_VIEW_CONTROLLER)
+        let appDelegate = (UIApplication.shared.delegate as! AppDelegate)
+        appDelegate.window?.rootViewController = viewController
+    }
+    
+    func onMobileRTCAuthReturn(_ returnValue: MobileRTCAuthError) {
```

Comment:
<img width="615" alt="Screenshot 2020-10-07 at 11 15 59 AM" src="https://user-images.githubusercontent.com/328112/95292152-926e1c00-088e-11eb-8630-a163f677438d.png">
<img width="623" alt="Screenshot 2020-10-07 at 11 16 25 AM" src="https://user-images.githubusercontent.com/328112/95292165-9863fd00-088e-11eb-9552-2dd8bdf6e249.png">

Reply:
`join` is a different abstraction than the rest of code.